src/plugins/unified_search/public/dataview_picker/change_dataview.tsx

```diff
@@ -262,7 +271,7 @@ export function ChangeDataView({
               setIsTextBasedLangSelected(false);
               // clean up the Text based language query
               onTextLangQuerySubmit?.({
-                language: 'kql',
+                language: 'kuery',
```

Comment:
ℹ️  This is called on cleanup (i.e. when navigating from the text based languages mode to the dataviewMode. I was wrongly setting the language to kql while it is kuery